connection.go

```diff
@@ -443,6 +443,12 @@ func (sc *snowflakeConn) Ping(ctx context.Context) error {
 // CheckNamedValue determines which types are handled by this driver aside from
 // the instances captured by driver.Value
 func (sc *snowflakeConn) CheckNamedValue(nv *driver.NamedValue) error {
+	switch nv.Value.(type) {
```

Comment:
maybe this (since you only check one type):

```
if _, ok := nv.Value.(SnowflakeDataType); ok {
   return nil
}
```

Reply:
Oh nice - it did seem a little clunky to have to do a type switch just to check a single type.